### PR TITLE
config: new interpolation function jsonencode

### PR DESCRIPTION
**Pull request by glasser**
We find ourselves wanting to construct JSON config files in (eg) S3 bucket objects based on the values of Terraform variables, so being able to JSON-encode strings safely is helpful.

### CODE REVIEW

File: config/interpolate_funcs_test.go [hunk 1]
```diff
@@ -437,6 +437,38 @@ func TestInterpolateFuncJoin(t *testing.T) {
...
+			{
+				`${jsonencode(easy)}`,
+				`"test"`,
+				false,
+			},
+			{
+				`${jsonencode(hard)}`,
+				`" foo \\ \n \t \" bar "`,
+				false,
+			},
```
Comment (radeksimko) [inline config/interpolate_funcs_test.go]: Would you mind adding one more test case with an empty string to confirm that an error is returned in such cases?

**Comment by radeksimko** [comment]
Besides the test case w/ failure, this is looking good. It has other good test cases + documentation. :+1:

File: config/interpolate_funcs_test.go (hunk 1 above)
Comment (glasser) [inline config/interpolate_funcs_test.go]: I can add a test case for the empty string, but why should it be an error rather than `""`?

Comment (glasser) [inline config/interpolate_funcs_test.go]: I added a test case for the empty string (no error) and for no argument given (error).

Comment (radeksimko) [inline config/interpolate_funcs_test.go]: Empty string should remain empty string, naturally. :+1: I was more looking for any test covering the erroneous state.

**Comment by glasser** [comment]
Thanks!
